game-core/src/main/java/games/strategy/triplea/ui/MapUnitTooltipManager.java

```diff
@@ -0,0 +1,58 @@
+package games.strategy.triplea.ui;
+
+import java.awt.MouseInfo;
+import java.awt.Point;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.JToolTip;
+import javax.swing.Popup;
+import javax.swing.PopupFactory;
+import javax.swing.Timer;
+
+
+/**
+ * Responsible for showing tool tips when hovering over units on the main map.
+ */
+public class MapUnitTooltipManager implements ActionListener {
+  private final MapPanel mapPanel;
+  private Timer timer;
+  private String text;
+  private Popup popup;
+
+  public MapUnitTooltipManager(MapPanel mapPanel) {
+    this.mapPanel = mapPanel;
+    this.timer = new Timer(1000, this);
+    this.timer.setRepeats(false);
+    // Note: Timer not started yet.
+  }
+
+  /** Updates the tooltip. */
```

Comment:
Maybe it's just me, but I'd appreciate if this javadoc could be expanded to the "normal" 3 lines